### ISSUE TITLE
Feat: Add "Follow" button to profile

### ISSUE THREAD

**Issue by gbonaspetti** [tracim/tracim#3910]
## Feature description and goals

- [ ] If User X is viewing User Y's profile and X is not following Y and Y is not following X, the button will be labeled "Follow".
- [ ] If User X is viewing User Y's profile and X is not following Y, but Y is following  X, the button will be labeled "Follow back".
- [ ] If User X is viewing User Y's profile and X is following  Y, the button will be labeled "Unfollow".
- [ ] On click, during the "Follow" and "Follow back" states, the user starts to follow the other one and the button will change to the "Unfollow" state.
- [ ] On click, during the "Unfollow" state, the user stop to follow the other one and the button changes to the "Follow" or "Follow back" state.
- [ ] The button is hidden if the profile is of the logged user.

## Translations

|English|French|Portuguese|
|-|-|-|
|Follow|S'abonner|Seguir|
|Follow back|Suivre en retour|Seguir de volta|
|Unfollow|Se desabonner|Deixar de seguir|

## Interface

![public_profile](https://user-images.githubusercontent.com/35422692/100859848-0e03d800-3490-11eb-990d-9d68e3db75ac.jpg)

## Prerequisites

#3829

**Comment by grignards** [comment]
I think this is blocked by the backend as this relationship as to be stored.

**Comment by grignards** [comment]
Is there a different semantic between "follow" and "follow back"? This text difference makes me think that there will be, but I do not see which?

**Comment by gbonaspetti** [comment]
> Is there a different semantic between "follow" and "follow back"? This text difference makes me think that there will be, but I do not see which?

"follow back" means that you will follow each other mutually and all functionalities that are only for users connected to each other will be available. "follow" means that the other person is not following you yet and you will have to wait for them to follow you back in order to have access to those functionalities.

**Comment by grignards** [comment]
Concerning the translation: shouldn't it be "suivre" in french?

**Comment by grignards** [comment]
> "follow back" means that you will follow each other mutually and all functionalities that are only for users connected to each other will be available. "follow" means that the other person is not following you yet and you will have to wait for them to follow you back in order to have access to those functionalities.

Yes I understood the "back" word, what i miss(-ed) are "those functionalities". In between I remembered the private messaging which will (?) be linked to this follow.

**Comment by gbonaspetti** [comment]
> Concerning the translation: shouldn't it be "suivre" in french?

I followed the models in the translations, can you tell me if "suivre" is used in this context? In Portuguese is follow (seguir) too.

**Comment by gbonaspetti** [comment]
> > "follow back" means that you will follow each other mutually and all functionalities that are only for users connected to each other will be available. "follow" means that the other person is not following you yet and you will have to wait for them to follow you back in order to have access to those functionalities.
> 
> Yes I understood the "back" word, what i miss(-ed) are "those functionalities". In between I remembered the private messaging which will (?) be linked to this follow.

for the moment we don't have different functionalities, in the future the user will be able to chat only with the people he follows and some ideas for personalizing the personal feed also turn around the following

**Comment by raphj** [comment]
Yes, I think we can say "suivre". "S'abonner" ("subscribe") would be more suitable for a channel or a feed.
We cannot really "S'abonner" to somebody
